### PR TITLE
feat(meta): Create distributions meta tables

### PR DESCRIPTION
**Pull request by evanh**
This creates the local table, distributed table and materialized view for the
meta tables of the distributions cluster.

Depends on https://github.com/getsentry/snuba/pull/5759
Depends on https://github.com/getsentry/snuba/pull/5749

### CODE REVIEW

File: snuba/snuba_migrations/generic_metrics/0036_distributions_meta_tables.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+                    ttl="timestamp + toIntervalDay(retention_days)",
+                ),
+                columns=self.meta_table_columns,
+                target=OperationTarget.LOCAL,
+            ),
+            operations.CreateTable(
+                storage_set=self.storage_set_key,
+                table_name=self.meta_dist_table_name,
+                engine=table_engines.Distributed(
+                    local_table_name=self.meta_local_table_name, sharding_key=None
```
Comment (volokluev) [inline snuba/snuba_migrations/generic_metrics/0036_distributions_meta_tables.py:63]: dist table name?

File: snuba/snuba_migrations/generic_metrics/0036_distributions_meta_tables.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+                    ttl="timestamp + toIntervalDay(retention_days)",
+                ),
+                columns=self.tag_value_table_columns,
+                target=OperationTarget.LOCAL,
+            ),
+            operations.CreateTable(
+                storage_set=self.storage_set_key,
+                table_name=self.tag_value_dist_table_name,
+                engine=table_engines.Distributed(
+                    local_table_name=self.tag_value_local_table_name, sharding_key=None
```
Comment (volokluev) [inline snuba/snuba_migrations/generic_metrics/0036_distributions_meta_tables.py:115]: dist table name?

File: snuba/snuba_migrations/generic_metrics/0036_distributions_meta_tables.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        Column("project_id", UInt(64)),
+        Column("metric_id", UInt(64)),
+        Column("tag_key", UInt(64)),
+        Column("tag_value", String()),
+        Column("timestamp", DateTime(modifiers=Modifiers(codecs=["DoubleDelta"]))),
+        Column("retention_days", UInt(16)),
+        Column("count", AggregateFunction("sum", [Float(64)])),
+    ]
+
+    storage_set_key = StorageSetKey.GENERIC_METRICS_SETS
```
Comment (nikhars) [inline snuba/snuba_migrations/generic_metrics/0036_distributions_meta_tables.py:41]: Please fix the storage set key

File: snuba/snuba_migrations/generic_metrics/0036_distributions_meta_tables.py (hunk 1 above)
Comment (evanh) [reply snuba/snuba_migrations/generic_metrics/0036_distributions_meta_tables.py:63]: This setting is telling the Distributed engine which local table to use as the storage node. You can see table_name above has the dist table in it.

**Review by nikhars — APPROVED** [review]
Approving the changes. Please fix the ordering of the migrations list as you merge them. Since there would be conflicts in the group loader definition.

**Pull request merged**